frontend/src/js/components/editors/template/TemplateLabelsEditor.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import TemplateLabelSettings from './TemplateLabelSettings.jsx'
+
+const labels = ["dataname", "title", "units", "mean", "min", "max", "file", "crdate", "crtime", "zname", "tname", "zvalue", "tvalue", "zunits", "tunits"]
+
+
+var TemplateLabelsEditor = React.createClass({
+    propTypes: {
+        template: React.PropTypes.object,
+        updateTemplate: React.PropTypes.func
+    },
+    updateTemplateAttribute(attribute) {
+        const self = this;
+        return (key, value) => {
+            self.props.updateTemplate(attribute, key, value);
+        }
+    },
+    getInitialState() {
+        return {workingTemplate: $.extend({}, this.props.template)};
+    },
+    componentWillReceiveProps(nextProps) {
+        this.setState({workingTemplate: $.extend({}, nextProps.template)});
+    },
+    render() {
+        const template = this.state.workingTemplate;
+        const self = this;
```

Comment:
your code additions seem to work fine now how do you want me to wire up the save and cancel buttons (see below)